posts/inside-rust/2021-03-03-planning-rust-2021.md

```diff
@@ -0,0 +1,68 @@
+
+The final list of features for inclusion in Rust 2021 is still being decided. Overall, we aim for Rust 2021 to be a much smaller release than Rust 2018. This is for several reasons: 
+* establishing a regular cadence for edition releases means we get many of the benefits of Rust's "train" release model at the edition level.
+* Rust 2018 worked directly against the Rust model of "low stress" releases. 
+* there's simply fewer breaking changes needed to continue to evolve the language. 
+
+You can read more about the evolution of the concept of editions [in the RFC](https://github.com/rust-lang/rfcs/pull/3085).
+
+Whether a feature will be included in Rust 2021 is a part of the RFC process, so the short list of possible features can and will change between now and the edition being released. That being said, here are some of the possible features that may be a part of the edition:
+
+### Prelude changes 
+
+While types and free functions can be added to the prelude independent of edition boundaries, the same is not true for traits. Adding a trait to the prelude can cause compatibility issues because calls to methods named the same as methods of the newly in-scope traits can become ambiguous. 
+
+Currently the following traits are being proposed for inclusion in the Rust 2021 edition:
+* `TryFrom`/`TryInto`
+* `FromIterator`
+
+The RFC for this change can be found [here](https://github.com/rust-lang/rfcs/pull/3090). Please note that the RFC is not yet merged, and the contents for a new prelude are still under active discussion.
+
+### New closure capture rules 
+
+[RFC 2229](https://github.com/rust-lang/rfcs/pull/2229) proposed that closures capture individual fields and not the whole struct when possible. This RFC has been accepted. In some circumstances this change would cause destructors to run at different times than they currently do, so the change must be tied to an edition. Migration lints will be provided to avoid changing the semantics of existing code.
+
+### New default feature resolver in Cargo 
+
+In Rust 1.51, Cargo will stabilize a new [feature resolver](https://github.com/rust-lang/cargo/issues/8088) which allows a crate's dependencies to use different features in different contexts. For example, a `#[no_std]` crate might want to use a particular dependency both as a build dependency (with `std` enabled) and as a regular dependency (with `std` disabled). Currently, this leads to `std` being enabled in both cases since features belong to a global namespace. 
+
+In Rust 2021 this new resolver will become the default, but older editions can still use the new resolver by opting into it.
+
+### Other 
+
+Other proposed changes include [unifying how `panic` in `std` and `core` work](https://github.com/rust-lang/rust/issues/80162), upgrading several lints from [warnings to errors](https://github.com/rust-lang/rust/issues/80165), and [removing `static mut`](https://github.com/rust-lang/rust/issues/53639). 
```

Comment:
I agree with the comments in the linked issue about `static mut`: we need a concrete proposal in place before we consider deprecating `static mut`, and we don't have such a proposal yet. Given that, I don't think we should mention it here.